python_modules/libraries/dagster-aws/dagster_aws/redshift/resources.py

```diff
@@ -203,9 +207,33 @@ def _get_cursor(self, conn, cursor_factory=None):
                 conn.commit()
 
 
-class FakeRedshiftResource(_BaseRedshiftResource):
+@deprecated
+class RedshiftResource(RedshiftClient):
+    pass
+
+
+class FakeRedshiftClient(BaseRedshiftClient):
     QUERY_RESULT = [(1,)]
 
+    def __init__(self, context: InitResourceContext):
+        # Extract parameters from resource config
+        self.conn_args = {
+            k: context.resource_config.get(k)
+            for k in (
+                "host",
+                "port",
+                "user",
+                "password",
+                "database",
+                "connect_timeout",
+                "sslmode",
+            )
+            if context.resource_config.get(k) is not None
+        }
+
+        self.autocommit = context.resource_config.get("autocommit")
+        self.log = context.log
+
```

Comment:
Yeah I didn't mean to move you to do a big copy paste job.

I think the core issue here is that the stuff you are porting aren't really "fakes".

Reply:
I believe you can just delete `self.conn_args` and `self.autocommit`?